Source/engine/point.hpp

```diff
@@ -91,17 +68,22 @@ struct Point {
 		return a;
 	}
 
-	constexpr friend Point operator-(Point a, const Point &b)
+	constexpr friend Displacement operator-(Point a, const Point &b)
 	{
-		a -= b;
-		return a;
+		return { b.x - a.x, b.y - a.y };
 	}
 
 	constexpr friend Point operator-(const Point &a)
 	{
 		return { -a.x, -a.y };
 	}
 
+	constexpr friend Point operator-(Point a, Displacement displacement)

```

Comment:
How about + for symmetry?

Reply:
I believe this is deliberate. A `Displacement` represents the difference between two `Points` so it is only well defined for subtraction. Combining (adding) two points is rare and I think it would be better expressed as `displacementFromOrigin = (endPoint - origin); startPoint + displacementFromOrigin;`.

Shortcuts could be provided using `Displacement::Displacement(Point)` or a similar conversion from a `Point` to a `Displacement` implied to be from the origin if it turns out to be useful.